### PR TITLE
Goal: add comments

### LINKED ISSUE #92: Goal: add comments
- `Comment` component: view/edit mode
- add new comment form to Goal page
- highlight new comments

Figma https://www.figma.com/file/OZhSt60CA5rinmbr5aTfSP/Taskany?node-id=945%3A642

### PR DESCRIPTION
**Pull request by peter-nikitin**
Resolve #92

### CODE REVIEW

File: graphql/resolvers/Comment.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            const [commentAuthor, goal, commentsCount] = await Promise.all([
+                db.user.findUnique({ where: { id: user.id }, include: { activity: true } }),
+                db.goal.findUnique({ where: { id: goalId } }),
+                db.comment.count(),
+            ]);
+
+            if (!commentAuthor) return null;
+            if (!goal) return null;
+
+            const newCommentId = `${goal?.id}-${commentsCount + 1}`;
```
Comment (awinogradov) [inline graphql/resolvers/Comment.ts:48]: We don't need readable id for Comment.

File: src/components/Comment.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        border-radius: 2px;
+        z-index: 3;
+        transform: rotate(-45deg);
+        top: 14px;
+        left: -10px;
+    }
+`;
+
+const StyledTip = styled(Tip)`
+    padding: 0 !important;
```
Comment (awinogradov) [inline src/components/Comment.tsx:69]: Why you need `!important` here?

File: src/components/Comment.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+import { FormTextarea } from './FormTextarea';
+import { UserPic } from './UserPic';
+import { Icon } from './Icon';
+import { Text } from './Text';
+import { FormAction } from './FormActions';
+import { Button } from './Button';
+import { Tip } from './Tip';
+import { Link } from './Link';
+
+interface CommentsCreateFormProps {
```
Comment (awinogradov) [inline src/components/Comment.tsx:25]: `CommentProps` https://github.com/taskany-inc/issues/blob/main/docs/patterns.md#no-abstract-names-for-interfaces

File: src/components/FormTextarea.tsx [hunk 4]
```diff
@@ -17,6 +17,7 @@ interface FormTextareaProps {
     disabled?: boolean;
     style?: React.CSSProperties;
     flat?: 'top' | 'bottom' | 'both';
+    minHeight?: number;
```
Comment (awinogradov) [inline src/components/FormTextarea.tsx:20]: Why not with `styled`?

File: src/components/GlobalStyle.ts [hunk 5]
```diff
@@ -17,4 +17,8 @@ export const GlobalStyle = createGlobalStyle`
 
         background-color: ${backgroundColor};
     }
+
+    * {
```
Comment (awinogradov) [inline src/components/GlobalStyle.ts:21]: No way) Don't use global overrides plz

**Review by awinogradov — COMMENTED** [review]
- you need to add connections between Goal and Comment in prisma.schema
- you need to add https://github.com/taskany-inc/issues/blob/main/graphql/resolvers/Goal.ts#L90-L110

File: src/components/Comment.tsx (hunk 2 above)
Comment (peter-nikitin) [reply src/components/Comment.tsx:69]: remove

File: src/components/Comment.tsx (hunk 3 above)
Comment (peter-nikitin) [reply src/components/Comment.tsx:25]: fix

File: src/components/GlobalStyle.ts (hunk 5 above)
Comment (peter-nikitin) [reply src/components/GlobalStyle.ts:21]: move this style to FormInput and FormTextarea

File: src/components/FormTextarea.tsx (hunk 4 above)
Comment (peter-nikitin) [reply src/components/FormTextarea.tsx:20]: Right, not a good approach 
Change to `StyledFormTextarea` in `Commnet`

File: graphql/resolvers/Comment.ts (hunk 1 above)
Comment (peter-nikitin) [reply graphql/resolvers/Comment.ts]: change to nanoid

File: src/components/Comment.tsx [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+const StyledTip = styled(Tip)`
+    padding: 0;
+`;
+
+const StyledFormTextarea = styled(FormTextarea)`
+    min-height: 100px;
+`;
+
+export const Comment: React.FC<CommentProps> = ({ onCreate, goalId }) => {
+    const { data: session } = useSession();
```
Comment (awinogradov) [inline src/components/Comment.tsx:77]: Do it with external prop `user`

Comment (peter-nikitin) [reply src/components/Comment.tsx]: done

**Pull request merged**